src/server.ts

```diff
@@ -1,47 +1,98 @@
 import * as fsPath from 'path';
+import * as http from 'http';
 import * as nunjucks from 'nunjucks';
 
 import {Pod} from './pod';
 import {StaticRoute} from './router';
+import {Watcher} from './watcher';
 
 import express = require('express');
 
 export class Server {
-  constructor() {}
-}
+  private pod: Pod;
+  private app?: express.Application;
+  private httpServer?: http.Server;
+  private watcher: Watcher;
+  private port: string | number;
+
+  constructor(pod: Pod, port: string | number) {
+    this.pod = pod;
+    this.port = port;
+    this.watcher = new Watcher(pod, this);
+    this.app = undefined;
+    this.httpServer = undefined;
+  }
 
-export function createApp(pod: Pod) {
-  const app = express();
-  app.disable('x-powered-by');
-  pod.plugins.trigger('createServer', app);
-  app.all('/*', async (req: express.Request, res: express.Response) => {
-    try {
-      const route = pod.router.resolve(req.path);
-      if (!route) {
-        res
-          .status(404)
-          .sendFile(fsPath.join(__dirname, './static/', 'error-no-route.html'));
-        return;
+  /**
+   * Returns the `Express` server application.
+   */
+  createApp() {
+    const app = express();
+    this.pod.plugins.trigger('createServer', app);
+    app.disable('x-powered-by');
+    app.all('/*', async (req: express.Request, res: express.Response) => {
+      try {
+        const route = this.pod.router.resolve(req.path);
+        if (!route) {
+          res
+            .status(404)
+            .sendFile(
+              fsPath.join(__dirname, './static/', 'error-no-route.html')
+            );
+          return;
+        }
+        if (route.provider.type === 'static_dir') {
+          res.sendFile(
+            this.pod.getAbsoluteFilePath(
+              (route as StaticRoute).staticFile.podPath
+            )
+          );
+          return;
+        } else {
+          const content = await route.build();
+          res.set('Content-Type', route.contentType);
+          res.send(content);
+        }
+      } catch (err) {
+        nunjucks.configure(fsPath.join(__dirname, './static/'), {
+          autoescape: true,
+          express: app,
+        });
+        res.status(500).render('error-generic.njk', {
+          error: err,
+        });
       }
-      if (route.provider.type === 'static_dir') {
-        res.sendFile(
-          pod.getAbsoluteFilePath((route as StaticRoute).staticFile.podPath)
+    });
+    return app;
+  }
+
+  /**
+   * Starts the web server.
+   */
+  start(startWatcher = false) {
+    this.app = this.createApp();
+    this.httpServer = this.app.listen(this.port, () => {
+      // Only output first time server is started.
+      if (startWatcher) {
```

Comment:
maybe this is a problem with readability, but why is this only logged when startWatcher is true? i'm assuming there's some logic that says "the watcher is only started the first time the server is started, when the first time the server is started, log the host/port", but it's not clear here from just reading this code.

Reply:
Agree this was a bit overloaded; I was thinking so myself. I moved this out to explicit options for `log` and `watch`.